### PR TITLE
Sounds

### PR DESCRIPTION
**Pull request by notV4l**
- #35 #47 
- add sound toggle in header (removed)
- add headerButton component
- add a simple store to play sounds (and some test sounds 🎵 )

- add media player
- add mobile menu
- customizable hover/click sound for Buttons

### CODE REVIEW

**Comment by broody** [comment]
Looks great! Some small feedback

File: web/src/hooks/sound.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
+import { create } from "zustand";
+// import { persist, createJSONStorage } from 'zustand/middleware'
```
Comment (tarrencev) [inline web/src/hooks/sound.tsx:2]: remove?

**Comment by broody** [comment]
It looks like build is failing on this

```
./src/components/HeaderButton.tsx:11:21
--
13:06:45.349 | Type error: Parameter 'size' implicitly has an 'any' type.
13:06:45.349 |  
13:06:45.349 | 9 \| import { IsMobile } from "@/utils/ui";
13:06:45.349 | 10 \|
13:06:45.349 | > 11 \| const clipCorner = (size) => `polygon(${size} 0%, calc(100% - ${size}) 0%, 100% ${size}, 100% calc(100% - ${size}), calc(100% - ${size}) 100%, ${size} 100%, 0% calc(100% - ${size}), 0% ${size})`

```

**Comment by tarrencev** [comment]
Hey @notV4l once the feedback has been addressed and ci is passing we can get this merged!

**Comment by tarrencev** [comment]
> > Hey @notV4l once the feedback has been addressed and ci is passing we can get this merged!
> 
> i'll fix while doing #47 i can't access CI logs/issues btw :(

Ah ok! I think if you do `yarn build` locally you will see the ci issues. Lets fix here before #47, so we can merge this

File: web/src/pages/index.tsx [hunk 2]
```diff
@@ -44,20 +49,29 @@ export default function Home() {
               joined={5}
               max={6}
               onClick={() => router.push("/0xdead")}
+              onMouseEnter={()=> {
```
[the remark this answers is not in the document]
Comment (notV4l) [reply web/src/pages/index.tsx]: done

File: web/src/pages/index.tsx [hunk 3]
```diff
@@ -30,7 +32,10 @@ export default function Home() {
             <Text color="neon.500" fontSize="14px">
               YOU HAVE NO GAMES
             </Text>
-            <Button w="full" onClick={() => router.push("/create")}>
+            <Button w="full"
```
[the remark this answers is not in the document]
Comment (notV4l) [reply web/src/pages/index.tsx]: done

File: web/src/components/Header.tsx [hunk 4]
```diff
@@ -56,19 +76,37 @@ const Header = () => {
...
-          p="8px"
-          layerStyle="rounded"
-          cursor="pointer"
-          onClick={() => setConnected(true)}
-        >
-          <Link /> <Text>CONNECT</Text>
+        <HStack flex="1" justify="right">
+          {/* <HeaderButton onClick={() => toggleIsMuted()}>
+            <Sound isMuted={isMuted} />
+          </HeaderButton> */}
```
Comment (tarrencev) [inline web/src/components/Header.tsx:93]: remove?

File: web/src/hooks/media.tsx [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+  useMediaStore.setState((state) => ({
+    currentIndex: (state.currentIndex + 1) % state.medias.length,
+  }));
+
+  state.isPlaying && play();
+};
+
+// // Change global volume.
+// Howler.volume(0.5);
```
Comment (tarrencev) [inline web/src/hooks/media.tsx:113]: remove?

File: web/src/hooks/sound.tsx [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+  if (state.isInitialized) return;
+  const context = new AudioContext();
+
+  // for (let sound of Object.keys(Sounds)) {
+  //   state.library.buffers[sound] = await loadSoundBuffer(
+  //     `/sounds/${Sounds[sound as keyof Sounds]}`,
+  //     context,
+  //   );
+  // }
+
```
Comment (tarrencev) [inline web/src/hooks/sound.tsx:56]: remove?

File: web/src/hooks/sound.tsx [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+  library.sources[sound] = source;
+  library.gains[sound] = gainNode;
+
+  // useSoundStore.setState( (state) => {
+  //   // let s = await state;
+  //   let s = state;
+  //   s.library.sources[sound] = source;
+  //   s.library.gains[sound] = gainNode;
+  // });
```
Comment (tarrencev) [inline web/src/hooks/sound.tsx:107]: remove?

Comment (notV4l) [reply web/src/hooks/sound.tsx:107]: do u hate comments XD

Comment (tarrencev) [reply web/src/hooks/sound.tsx]: haha otherwise they sit around forever

**Pull request merged**